src/js/playlist/source.js

```diff
@@ -31,6 +32,13 @@ define([
         if (! _source.type) {
```

Comment:
down below maybe we should handle the three cases where we set type in else if blocks ~like:
```
if youtube
else if rtmp
else
  check extension
```